### PR TITLE
[HttpClient] Don't read from the network faster than the CPU can deal with

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.3
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Something I spotted while working on #35115: both the curl and native clients don't play well with heavily compressed HTTP streams: they decompress faster than userland can process chunks.

The attached patch moves the decompression logic to the chunk generator. This means internally we only deal with raw compressed chunks, and they are decompressed only when passing the value to userland.

### CODE REVIEW

File: src/Symfony/Component/HttpClient/CurlHttpClient.php [hunk 1]
```diff
@@ -113,7 +113,7 @@ public function request(string $method, string $url, array $options = []): Respo
         $url = implode('', $url);
 
         if (!isset($options['normalized_headers']['user-agent'])) {
-            $options['normalized_headers']['user-agent'][] = $options['headers'][] = 'User-Agent: Symfony HttpClient/Curl';
+            $options['headers'][] = 'User-Agent: Symfony HttpClient/Curl';
```
Comment (stof) [inline src/Symfony/Component/HttpClient/CurlHttpClient.php]: is this change related to this gzip issue ?

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpClient/CurlHttpClient.php]: not directly

File: src/Symfony/Component/HttpClient/Response/ResponseTrait.php [hunk 2]
```diff
@@ -299,6 +317,16 @@ public static function stream(iterable $responses, float $timeout = null): \Gene
                         $isTimeout = false;
 
                         if (\is_string($chunk = array_shift($multi->handlesActivity[$j]))) {
+                            if (null !== $response->inflate && false === $chunk = @inflate_add($response->inflate, $chunk)) {
+                                $multi->handlesActivity[$j] = [null, new TransportException('Error while processing content unencoding.')];
+                                continue;
+                            }
+
+                            if ('' !== $chunk && null !== $response->content && \strlen($chunk) !== fwrite($response->content, $chunk)) {
+                                $multi->handlesActivity[$j] = [null, new TransportException('Failed writing %d bytes to the response buffer.', \strlen($chunk))];
```
Comment (TiGR) [inline src/Symfony/Component/HttpClient/Response/ResponseTrait.php]: @nicolas-grekas I think `sprintf` is supposed to be here.

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpClient/Response/ResponseTrait.php]: correct :) would you mind sending a PR?

Comment (TiGR) [reply src/Symfony/Component/HttpClient/Response/ResponseTrait.php]: Done #35250
